modin/core/execution/client/query_compiler.py

```diff
@@ -35,7 +35,7 @@ def __init__(self, id):
         self._id = id
 
     def _set_columns(self, new_columns):
-        self._id = self._service.rename(self._id, new_col_labels=new_columns)
+        self._service.set_columns(self._id, new_columns)
```

Comment:
I wonder if @devin-petersohn would understand the motivations for this decision better, but in general, I've always assumed that most dataframe metadata modifications would result in a new query compiler instead of modifying the attribute directly. I assumed that this was intended by design to keep the QC immutable.

Reply:
@pyrito it's true that the almost every query compiler method creates a new compiler, but as I said in the commit description, `index` and `columns` are mutable properties of every query compiler, including the one we're using in our service right now (not naming it since we're in OSS). However, our service doesn't expose an interface to update one of the backend query compilers in place. I don't know a good reason right now to not expose such an interface. Maybe in the future we'd want to cache service results by query compiler ID, in which case mutating a backend query compiler would not be allowed.

Reply:
Can we add a comment for this?

Reply:
Done.